kasa/smartcamera/modules/alarm.py

```diff
@@ -0,0 +1,168 @@
+                container=self,
+                attribute_getter="active",
+                icon="mdi:bell",
+                type=Feature.Type.BinarySensor,
+            )
+        )
+        self._add_feature(
+            Feature(
+                device,
+                id="alarm_sound",
+                name="Alarm sound",
+                container=self,
+                attribute_getter="alarm_sound",
+                attribute_setter="set_alarm_sound",
+                category=Feature.Category.Config,
+                type=Feature.Type.Choice,
+                choices_getter="alarm_sounds",
+            )
+        )
+        self._add_feature(
+            Feature(
+                device,
+                id="alarm_volume",
+                name="Alarm volume",
+                container=self,
+                attribute_getter="alarm_volume",
+                attribute_setter="set_alarm_volume",
+                category=Feature.Category.Config,
+                type=Feature.Type.Number,
+                range_getter=lambda: (VOLUME_MIN, VOLUME_MAX),
+            )
+        )
+        self._add_feature(
+            Feature(
+                device,
+                id="alarm_duration",
+                name="Alarm duration",
+                container=self,
+                attribute_getter="alarm_duration",
+                attribute_setter="set_alarm_duration",
+                category=Feature.Category.Config,
+                type=Feature.Type.Number,
+                range_getter=lambda: (DURATION_MIN, DURATION_MAX),
+            )
+        )
+        self._add_feature(
+            Feature(
+                device,
+                id="test_alarm",
+                name="Test alarm",
+                container=self,
+                attribute_setter="play",
+                type=Feature.Type.Action,
+            )
+        )
+        self._add_feature(
+            Feature(
+                device,
+                id="stop_alarm",
+                name="Stop alarm",
+                container=self,
+                attribute_setter="stop",
+                type=Feature.Type.Action,
+            )
+        )
+
+    @property
+    def alarm_sound(self) -> str:
+        """Return current alarm sound."""
+        return self.data["getSirenConfig"]["siren_type"]
+
+    async def set_alarm_sound(self, sound: str) -> dict:
+        """Set alarm sound.
+
+        See *alarm_sounds* for list of available sounds.
+        """
+        if sound not in self.alarm_sounds:
+            msg = f"sound must be one of {', '.join(self.alarm_sounds)}: {sound}"
+            raise ValueError(msg)
+        return await self.call("setSirenConfig", {"siren": {"siren_type": sound}})
+
+    @property
+    def alarm_sounds(self) -> list[str]:
+        """Return list of available alarm sounds."""
+        return self.data["getSirenTypeList"]["siren_type_list"]
+
+    @property
+    def alarm_volume(self) -> Literal[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]:
```

Comment:
I'd just keep this as an int, and if really wanted, raise a suitable error if something out of range is tried in `set_alarm_volume`.

Reply:
Done